Assets/Scripts/Player/Interaction/Pluckable.cs

```diff
@@ -6,26 +6,68 @@
 public class Pluckable : NetworkBehaviour
 {
     [Tooltip("Vegetable/fruit spawned when you pluck")]
-    public NetworkPrefabRef prefabToPlant;
+    public NetworkPrefabRef[] prefabsToSpawn;
+
+    [Tooltip("Only pluckable if stageController says it is ripe")]
+    public bool useStageController = true;
 
     [Tooltip("Dependent on this being in final stage before being pluckable")]
     public StageController stageController;
 
+    [Tooltip("Pickup item produced if true, otherwise it drops on ground")]
+    public bool autoPickup = true;
+
+    [Tooltip("Spawns randomly from prefabsToPlant if true. Otherwise spawns ALL items in prefabsToPlant")]
+    public bool spawnRandomPrefab = false;
+
+    [Tooltip("Gameobject contains AudioSource's for when item is plucked")]
+    public GameObject pluckingNoises;
+
     public Pickupable toItem()
     {
-        if (!stageController.ripe)
+        if ((useStageController && !stageController.ripe) || prefabsToSpawn.Length < 1)
         {
             return null;
         }
-        var obj = Runner.Spawn(prefabToPlant, transform.position, transform.rotation);
+
+        NetworkObject obj;
+
+        if (spawnRandomPrefab)
+        {
+            var randomIndex = Random.Range(0, prefabsToSpawn.Length - 1);
+            obj = Runner.Spawn(prefabsToSpawn[randomIndex], transform.position, transform.rotation);
+        }
+        else
+        {
+            obj = Runner.Spawn(prefabsToSpawn[0], transform.position, transform.rotation); // return the first
+            spawnRestOfItems(prefabsToSpawn);
+        }
+
         StartCoroutine(KillMe()); // Async otherwise we can't return Pickupable
-        PlayAudio.PlayRandomSourceOnGameobject(gameObject);
+
+        if (pluckingNoises != null)
+        {
+            PlayAudio.PlayRandomSourceOnGameobject(pluckingNoises);
+        }

```

Comment:
I feel this method should just handle null values rather than us put guards everywhere we call it

Reply:
Agreed, it handles whether the GameObject has no AudioSources on it but not whether not the gameobject is null.

Programming without Option is really annoying.